src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,119 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
```

Comment:
The `Node` class should be an inner class within `MyLinkedList` to enhance encapsulation and maintainability. This change will help in keeping the `Node` class closely tied to the `MyLinkedList` implementation.